lib/train/file.rb

```diff
@@ -55,17 +53,21 @@ def type
     end
 
     def md5sum
-      res = Digest::MD5.new
-      res.update(content)
-      res.hexdigest
+      value = @backend.run_command("md5sum #{@path}").stdout.slice(0..31)
+      if value.empty?
+        value = nil
+      end
+      value
     rescue TypeError => _
```

Comment:
Are there any other classes that I should rescue from?  I was thinking of those in https://github.com/chef/train/blob/master/lib/train/errors.rb